extensions/wikia/MyHome/SpecialWikiActivity.class.php

```diff
@@ -8,151 +8,157 @@ class SpecialWikiActivity extends UnlistedSpecialPage {
-		$wgOut->setPageTitle(wfMsg('oasis-activity-header'));
+		$out->setPageTitle( $this->msg( 'oasis-activity-header' ) );
 
-		$template = new EasyTemplate(dirname(__FILE__).'/templates');
-		$template->set('data', $data['results']);
+		$template = new EasyTemplate( __DIR__ . '/templates' );
+		$template->set( 'data', $data['results'] );
 
-		$showMore = isset($data['query-continue']);
-		if ($showMore) {
-			$template->set('query_continue', $data['query-continue']);
+		$showMore = isset( $data['query-continue'] );
+		if ( $showMore ) {
+			$template->set( 'query_continue', $data['query-continue'] );
 		}
-		if (empty($data['results'])) {
-			$template->set('emptyMessage', wfMsgExt("myhome-activity-feed-empty", array( 'parse' )));
+		if ( empty( $data['results'] ) ) {
+			$template->set( 'emptyMessage', $this->msg( 'myhome-activity-feed-empty' )->parse() );
 		}
 
-		$template->set_vars(array(
+		$template->set_vars( [
 			'showMore' => $showMore,
 			'type' => $this->feedSelected,
-			'wgBlankImgUrl' => $wgBlankImgUrl,
-		));
+			'wgBlankImgUrl' => $this->wg->BlankImgUrl,
+		] );
 
-		$wgOut->addHTML($template->render('activityfeed.oasis'));
+		$out->addHTML( $template->render( 'activityfeed.oasis' ) );
 
 		// page header: replace subtitle with navigation
 		global $wgHooks;
-		$wgHooks['PageHeaderIndexAfterExecute'][] = array($this, 'addNavigation');
+		$wgHooks['PageHeaderIndexAfterExecute'][] = [ $this, 'addNavigation' ];
 
-		if ($wgUser->isAnon()) {
-			$this->getOutput()->setSquidMaxage( 3600 ); // 1 hour
-			$this->getOutput()->tagWithSurrogateKeys(
+		if ( $user->isAnon() ) {
+			$out->setSquidMaxage( 3600 ); // 1 hour
+			$out->getOutput()->tagWithSurrogateKeys(
 				MyHome::getWikiActivitySurrogateKey()
 			);
 		}
 
-		wfProfileOut(__METHOD__);
+		wfProfileOut( __METHOD__ );
 	}
 
 	/**
 	 * Replaces page header's subtitle with navigation for WikiActivity
 	 *
 	 * @author macbre
 	 */
-	function addNavigation(&$moduleObject, &$params) {
-		global $wgUser;
-		wfProfileIn(__METHOD__);
+	public function addNavigation( &$moduleObject, &$params ) {
+		wfProfileIn( __METHOD__ );
 
-		$template = new EasyTemplate(dirname(__FILE__).'/templates');
+		$template = new EasyTemplate( __DIR__ . '/templates' );
+		$isLoggedIn = $this->getUser()->isLoggedIn();
 
 		// RT #68074: show default view checkbox for logged-in users only
-		$showDefaultViewSwitch = $wgUser->isLoggedIn() && ($this->defaultView != $this->feedSelected);
+		$showDefaultViewSwitch = $isLoggedIn && ( $this->defaultView != $this->feedSelected );
 
-		$template->set_vars(array(
+		$template->set_vars( [
 			'classWatchlist' => $this->classWatchlist,
 			'defaultView' => $this->defaultView,
-			'loggedIn' => $wgUser->isLoggedIn(),
+			'loggedIn' => $isLoggedIn,
 			'showDefaultViewSwitch' => $showDefaultViewSwitch,
 			'type' => $this->feedSelected,
-		));
+			'typeMessage' => FeedRenderer::getTypeMessage( $this->feedSelected )->plain(), // only used as message parameter
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)